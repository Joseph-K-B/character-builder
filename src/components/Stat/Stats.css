.stat-controls{
    background-color: darkblue;
}