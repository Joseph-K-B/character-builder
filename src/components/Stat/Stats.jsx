/* eslint-disable jsx-a11y/no-redundant-roles */
import React from "react";
import '../Stat/Stats.css'

function Stats({
    catchPhrase,
    setCatchPhrase,
    handleClick
}) {
    return (
        <section className='stat-controls'>
            <input
                id='catchphrase'
                name='catchphrase'
                aria-label='catchphrase-input'
                data-testid='catchphrase-input' 
                type='text' 
                value={catchPhrase}
                onChange={(e) => setCatchPhrase(e.target.value)}    
            />
            <label htmlFor='catchphrase' />
            <button type='button' onClick={handleClick}>Submit</button>
        </section>
    )
}
export default Stats