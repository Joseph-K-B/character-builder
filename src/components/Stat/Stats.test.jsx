import { render, screen } from "@testing-library/react";
import Stats from "./Stats";

it('should render input and button for stats', () => {
    const {container} = render(<Stats />);
    const input = screen.getByTestId('catchphrase-input');
    const button = screen.getByRole('button');

    expect(input).toBeInTheDocument();
    expect(button).toBeInTheDocument();
    expect(container).toMatchSnapshot();
});