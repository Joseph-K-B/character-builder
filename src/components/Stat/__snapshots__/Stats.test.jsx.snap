// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render input and button for stats 1`] = `
<div>
  <section
    class="stat-controls"
  >
    <input
      aria-label="catchphrase-input"
      data-testid="catchphrase-input"
      id="catchphrase"
      name="catchphrase"
      type="text"
      value=""
    />
    <label
      for="catchphrase"
    />
    <button
      type="button"
    >
      Submit
    </button>
  </section>
</div>
`;
